### ISSUE TITLE
Could SwiftUI's @propertyWrapper be an alternative form of injection?

### ISSUE THREAD

**Issue by drekka** [Swinject/Swinject#419]
Been messing around with various bits of the latest beta and it occurred to be that an implementation of SwiftUI's @propertyWrapper could be used to access a container and inject a property. 

Any thoughts?

**Comment by nathanblamires** [comment]
I like the idea. I think it could make usage throughout the code more clear and concise.
E.g. `@Injected var analytics: AnalyticsService`

**Comment by jakubvano** [comment]
I have been playing with this idea as well, see #421.

Main issue that I see is passing `resolver` to the property wrapper - I haven't figured out better way than global variable 😞

**Comment by drekka** [comment]
Haven't tried it, but could the resolver (main Container) be passed via an `@Environment` reference?

So it would be a requirement to set the container into the environment of the main view, then all propertywrappers can access it.

Honestly don't know if that will work :-)

**Comment by pjechris** [comment]
Using a property wrapper would mean having Swinject inject those dependencies as well when doing unit test.

Doing that we would change Swinject from a dependency injection container to a Service Locator.

I would rather wait for annotations being available in Swift.

**Comment by jakubvano** [comment]
@pjechris I agree, this would result in a not testable code, but there is no requirement to use it in the entire codebase..
I could see this as a much more powerful replacement for the SwinjectStoryboard - one rarely unittests the UI layer, and it would enable the injection into other UI objects not only `UIViewController`

**Comment by jakubvano** [comment]
Looks like [this feature of property wrapers] is already implemented in latest Xcode-beta, which enables us to get a resolver instance from the enclosing type 🎉 

POC implemented [here](https://github.com/Swinject/Swinject/blob/3.0/Sources/3.0%20API/Retrieval.swift), in action [here](https://github.com/Swinject/Swinject/blob/3.0/Tests/3.0%20API/RetrievalSpec.swift).

**Comment by Zeta611** [comment]
Any updates?

**Comment by mpdifran** [comment]
I'm digging into this to see if we can get it officially released! Also looking at how this could factor into SwiftUI. Stay tuned for more updates here!
